controllers/release/adapter.go

```diff
@@ -373,6 +420,56 @@ func (a *adapter) createManagedPipelineRun(resources *loader.ProcessingResources
 	return pipelineRun, nil
 }
 
+// createTenantPipelineRun creates and returns a new tenant Release PipelineRun. The new PipelineRun will include owner
+// annotations, so it triggers Release reconciles whenever it changes. The Pipeline information and the parameters to it
+// will be extracted from the given ReleasePlan. The Release's Snapshot will also be passed to the release
+// PipelineRun.
+func (a *adapter) createTenantPipelineRun(releasePlan *v1alpha1.ReleasePlan, snapshot *applicationapiv1alpha1.Snapshot) (*tektonv1.PipelineRun, error) {
+	tenantPipelineParams := []tektonv1.Param{}
+	for _, param := range releasePlan.Spec.Pipeline.Params {
+		tenantPipelineParams = append(tenantPipelineParams, tektonv1.Param{
+			Name: param.Name,
+			Value: tektonv1.ParamValue{
+				Type:      tektonv1.ParamTypeString,
+				StringVal: param.Value,
+			},
+		})
+	}
+
+	pipelineRun, err := utils.NewPipelineRunBuilder(metadata.TenantPipelineType, releasePlan.Namespace).
+		WithAnnotations(metadata.GetAnnotationsWithPrefix(a.release, integrationgitops.PipelinesAsCodePrefix)).
+		WithFinalizer(metadata.ReleaseFinalizer).
+		WithLabels(map[string]string{
+			metadata.ApplicationNameLabel:  releasePlan.Spec.Application,
+			metadata.PipelinesTypeLabel:    metadata.TenantPipelineType,
+			metadata.ReleaseNameLabel:      a.release.Name,
+			metadata.ReleaseNamespaceLabel: a.release.Namespace,
+			metadata.ReleaseSnapshotLabel:  a.release.Spec.Snapshot,
+		}).
+		WithObjectReferences(a.release, releasePlan, a.releaseServiceConfig, snapshot).
```

Comment:
I'm not sure we want to pass the RSC to a tenant pipeline.

Reply:
removed.